modules/nextflow/src/main/groovy/nextflow/trace/WebLogObserver.groovy

```diff
@@ -77,6 +85,23 @@ class WebLogObserver implements TraceObserver{
     WebLogObserver(String url) {
         this.httpClient.setUrl(checkUrl(url))
         this.webLogAgent = new Agent<>(this)
+
+        /*
+        Make sure that:
+            1) Path objects are converted to String (or it throws a StackOverflow error)
+            2) Dates follow ISO 8601
+            3) Durations are in millisecond
+         */
+        this.generator = new JsonGenerator.Options()
+                .addConverter(Path) { Path p, String key ->
+                    p.toUriString()
+                }
+                .addConverter(Duration) { Duration d, String key ->
+                    d.durationInMillis
+                }
+                .dateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+                .timezone("UTC")
```

Comment:
There's an identic date format string later in the code, what about introducing a common constant in the `Const` object.